server/src/main/java/io/crate/execution/engine/collect/sources/ShardCollectSource.java

```diff
@@ -238,16 +238,8 @@ public void afterIndexShardCreated(IndexShard indexShard) {
              *
              * So we wrap the creation in a supplier to create the providers lazy
              */
-
-            // MapperService is null for closed indices
-            if (indexShard.mapperService() == null) {
```

Comment:
If we want to show the information in `sys.shards` for closed shards, we need include them to `shards`.